### PR TITLE
Add data retention to the choose plan page

### PR DESCRIPTION
**Pull request by vinibrsl**
This PR adds data retention as a benefit of the new V4 plans.

### CODE REVIEW

File: priv/plans_v4.json [hunk 1]
```diff
@@ -17,7 +18,9 @@
     "yearly_product_id":"857080",
     "site_limit":10,
     "team_member_limit":3,
-    "features":["goals"]
+    "features":["goals"],
+    "data_retention_in_years": 3
+
```
Comment (zoldar) [inline priv/plans_v4.json:23]:
```suggestion
```

File: priv/plans_v4.json [hunk 2]
```diff
@@ -57,7 +66,9 @@
     "yearly_product_id":"857084",
     "site_limit":10,
     "team_member_limit":3,
-    "features":["goals"]
+    "features":["goals"],
+    "data_retention_in_years": 3
+
```
Comment (zoldar) [inline priv/plans_v4.json:71]:
```suggestion
```

File: priv/plans_v4.json [hunk 3]
```diff
@@ -77,7 +90,9 @@
     "yearly_product_id":"857086",
     "site_limit":10,
     "team_member_limit":3,
-    "features":["goals"]
+    "features":["goals"],
+    "data_retention_in_years": 3
+
```
Comment (zoldar) [inline priv/plans_v4.json:95]:
```suggestion
```

File: priv/repo/migrations/20231208125624_add_data_retention_in_years_to_plans.exs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+defmodule Plausible.Repo.Migrations.AddDataRetentionInYearsToPlans do
```
Comment (zoldar) [inline priv/repo/migrations/20231208125624_add_data_retention_in_years_to_plans.exs:1]: Pehaps it will be safer to extract this migration to a separate PR and deploy and run it before deploying the rest?

**Review by ukutaht — APPROVED** [review]

File: lib/plausible/billing/plan.ex [hunk 5]
```diff
@@ -23,23 +23,24 @@ defmodule Plausible.Billing.Plan do
...
     field :monthly_cost
     field :monthly_product_id, :string
     field :yearly_cost
     field :yearly_product_id, :string
   end
 
-  @fields ~w(generation kind features monthly_pageview_limit site_limit team_member_limit volume monthly_cost monthly_product_id yearly_cost yearly_product_id)a
+  @fields ~w(generation kind features monthly_pageview_limit site_limit team_member_limit volume data_retention_in_years monthly_cost monthly_product_id yearly_cost yearly_product_id)a
+  @required @fields --
+              ~w(monthly_cost yearly_cost monthly_product_id yearly_product_id data_retention_in_years)a
```
[the remark this answers is not in the document]
Comment (vinibrsl) [reply lib/plausible/billing/plan.ex]: Thanks! Changed in a546d9f8d6179d290b900444c5f195f21950ae0c

File: priv/repo/migrations/20231208125624_add_data_retention_in_years_to_plans.exs (hunk 4 above)
Comment (vinibrsl) [reply priv/repo/migrations/20231208125624_add_data_retention_in_years_to_plans.exs:1]: This table is not actually used in production, just for analytical purposes, so I think it's ok to deploy it with these changes.
